gems/aws-sdk-core/lib/aws-sdk-core/credentials_provider_chain.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+module AWS::SDK::Core
+  class CredentialsProviderChain
+    def call(config)
```

Comment:
Theres no state in this - I'd lean towards making this a module with a self.call method instead.  Its confusing to me that it does not have any instance state (nor is it intended to) - its purely functional rather than an object.